src/wallet/wallet2.cpp

```diff
@@ -13046,6 +13055,77 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw, bool is_hex) const
+{
+  if (is_hex || m_export_format == ExportFormat::Raw)
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+
+  std::string h = HEX_OUTPUT_MAGIC + "\n";
+  h.append(epee::string_tools::buff_to_hex_nodelimer(raw));
+
+  // Split the long string into multiple lines, similarly to how openssl
+  // prints certificates.
+  for (size_t i = 63; i < h.length(); i += 64)
+  {
+    h.insert(i, 1, '\n');
+  }
+
+  h.insert(h.end(), '\n');
+
+  return epee::file_io_utils::save_string_to_file(path_to_file, h);
+}
+//----------------------------------------------------------------------------------------------------
+
+bool wallet2::load_from_file(const std::string& path_to_file, std::string& target_str,
+                             size_t max_size)
+{
+  std::string data;
+  bool r = epee::file_io_utils::load_file_to_string(path_to_file, data, max_size);
+  if (!r)
+  {
+    return false;
+  }
+
+  // Ignore any whitespace.
+  // We try to convert the version without whitespace from hex to binary.
+  // If that fails, we stick to the original data (hence the copy).
+  std::string without_whitespace(data);
+  without_whitespace.erase(
+    std::remove_if(
+      without_whitespace.begin(),
+      without_whitespace.end(),
+      [](char c){ return (c =='\r' || c =='\t' || c == ' ' || c == '\n'); }),
+    without_whitespace.end());
+
+  if (0 != without_whitespace.compare(0, HEX_OUTPUT_MAGIC.length(), HEX_OUTPUT_MAGIC))
```

Comment:
This check should be bumped up before the copy + string manipulation so that "raw" files are not being copied.

Reply:
The problem is that it's easy to add some leading whitespace or newlines when copy/pasting. We can of course strip all leading whitespace, then check for the magic string, and then strip all remaining whitespace, but it complicates the code quite a bit. 

Of course, copying a file in the memory needlessly is bad. However, from the UX perspective, searching for the path, and then pasting or typing it takes seconds. This copy takes something like milliseconds presumably. If this was some tight code in handling the blockchain, the story would be different of course.

In either case, if you disagree with my reasoning, let me know and I can change the code.

Reply:
There isn't much complication:
```c++
const auto magic = std::find_if_not(data.begin(), data.end(), boost::algorithm::is_space());
if (!boost::algorithm::starts_with(boost::make_iterator_range(magic, data.end()), HEX_OUTPUT_MAGIC))
{
  ...
}
data.erase(data.begin(), magic + HEX_OUTPUT_MAGIC.size());
```

Reply:
Okay you win :) . Updated.